pkg/bucketing/engine.go

```diff
@@ -124,7 +134,7 @@ func (b *Engine) GetModifications(visitorID string, context map[string]interface
 		}
 	}
 
-	for _, c := range b.config.Campaigns {
+	for _, c := range b.getConfig().Campaigns {
```

Comment:
You might want to consider assigning a local variable to b.Config() early in this method - otherwise this might change while you're iterating over it if the config is updated in the background.

Reply:
Yep nice catch, thanks